### PR TITLE
Order Details: show shipping label creation info row when the order is eligible

### PR DESCRIPTION
**Pull request by jaclync**
Part of #2975 

## Why

For users who are new to shipping label creation in mobile, we want to show an info link for [more details](https://user-images.githubusercontent.com/22608780/95947034-63febc80-0e0b-11eb-9fbd-bca696ed0793.png). This PR implemented the first part of the task - adding an info link row to order details if the order is eligible for shipping label creation.

### Pre-existing issues in `ImageAndTitleAndTextTableViewCell`

I noticed that footnote style wasn't configured correctly in `ImageAndTitleAndTextTableViewCell`, since the title label font is configured in multiple places. There's some tech debt in the cell to have different configuration styles https://github.com/woocommerce/woocommerce-ios/issues/3419, which I hope to address this sprint. For this PR, I fixed the title font style in https://github.com/woocommerce/woocommerce-ios/commit/0956a9ce7bd0113ee1b8141345cc75758eef7037.

### Showed shipping label creation info link row in `OrderDetailsDataSource`

- Added a new `Row.shippingLabelCreationInfo(showsSeparator: Bool)` enum case for the info link row
- Since there are two states where the info link is shown (whether "Mark Order Complete" CTA is visible), some associated values have to be used to configure cells up to design. This complicates the switch statement for cell configuration, lemme know if you know of a better way to unwrap the associated values!

### Order that is processing

- Go to the orders tab
- In the Processing tab, tap on an order --> an info link row should be shown below the "Mark Order Complete" CTA

### Order that is not processing and not refunded

- Go to the orders tab
- In the All Orders tab, tap on an order that is not processing and not refunded --> an info link row should be shown below the "Create Shipping Label" CTA

## Example screenshots

Notes for design reviewers: I didn't see design where "Mark Order Complete" CTA isn't visible, so I placed the info link below the "Create Shipping Label" CTA for now. Please lemme know if you have different thoughts!

\ | eligible for shipping label creation - processing order | eligible for shipping label creation - non-processing non-refunded order | not eligible for shipping label creation - processing order | not eligible for shipping label creation - non-processing non-refunded order
-- | -- | -- | -- | --
dark | Simulator Screen Shot - iPhone 11 - 2021-03-02 at 11 32 31 | Simulator Screen Shot - iPhone 11 - 2021-03-02 at 11 32 22 | Simulator Screen Shot - iPhone 11 - 2021-03-02 at 11 35 32 | Simulator Screen Shot - iPhone 11 - 2021-03-02 at 11 35 29
light | Simulator Screen Shot - iPhone 11 - 2021-03-02 at 11 33 00 | Simulator Screen Shot - iPhone 11 - 2021-03-02 at 11 32 50 | Simulator Screen Shot - iPhone 11 - 2021-03-02 at 11 35 43 | Simulator Screen Shot - iPhone 11 - 2021-03-02 at 11 35 40

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by ghost** [comment]
The design implementation looks great to me @jaclync however maybe we could name it something other than just 'details' do you think 'shipping details' would work @Garance91540

**Comment by jaclync** [comment]
> The design implementation looks great to me @jaclync however maybe we could name it something other than just 'details' do you think 'shipping details' would work @Garance91540

Hey @woocommerce/mobile-designers! The "Details" here actually means "order's product details" and feels quite redundant to me, since it just navigates to another screen of the same list of products as in order details:

order details | tapping "Details" in the Product(s) section
-- | --
Simulator Screen Shot - iPhone 11 - 2021-03-02 at 14 32 47 | Simulator Screen Shot - iPhone 11 - 2021-03-02 at 14 32 49

I'm not sure why we have this separate product list screen in the first place 🤔

File: WooCommerce/Classes/ViewModels/Order Details/OrderDetailsDataSource.swift
Comment (pmusolino) [inline WooCommerce/Classes/ViewModels/Order Details/OrderDetailsDataSource.swift]: Maybe it is superfluous to specify “on mobile device”?

File: WooCommerce/Classes/ViewRelated/ReusableViews/ButtonTableViewCell.swift
Comment (pmusolino) [inline WooCommerce/Classes/ViewRelated/ReusableViews/ButtonTableViewCell.swift:65]: Q: Is there a reason why this is no more `private`?

**Review by pmusolino — COMMENTED** [review]
Looks good to me @jaclync!
I just noticed an issue on big screens (like on iPhone 12 pro) where the label and the icon are not vertically aligned.

**Review by Garance91540 — APPROVED** [review]
Can't wait to see this button in the app for real!!

Everything looks good except as @jaclync mentioned we shouldn't be seeing the "Details" row in there. I'm assuming it's a legacy from when we had the "Details" screen on the Order screen?

https://d.pr/i/05913c

**Comment by jaclync** [comment]
> Everything looks good except as @jaclync mentioned we shouldn't be seeing the "Details" row in there. I'm assuming it's a legacy from when we had the "Details" screen on the Order screen?
https://d.pr/i/05913c

@Garance91540 yup let's remove this "Details" row! We already have an issue for this removal https://github.com/woocommerce/woocommerce-ios/issues/3279, which I'll work on after this PR.

**Comment by jaclync** [comment]
> I just noticed an issue on big screens (like on iPhone 12 pro) where the label and the icon are not vertically aligned.

@pmusolino I think this misalignment is from the dynamic type to support various font sizes. If you change the system font size, the icon and text might look aligned in some sizes? This is a tricky issue, since the text could be any number of lines and I haven't found a good way to vertically align the center of the icon and the **first line** of the text. We also have an existing issue for this https://github.com/woocommerce/woocommerce-ios/issues/3574 - I'll see how we can align this better separately!

File: WooCommerce/Classes/ViewRelated/ReusableViews/ButtonTableViewCell.swift
Comment (jaclync) [reply WooCommerce/Classes/ViewRelated/ReusableViews/ButtonTableViewCell.swift:65]: Before this PR, we only need to customize a `ButtonTableViewCell`'s bottom spacing to 0pt. However, in this PR we could be showing the default bottom spacing (20pt) or 0pt in `OrderDetailsDataSource.configureMarkCompleteButton`. I thought it'd be better using the same default value from the cell in case we change it in the future. Lemme know if you have other thoughts!

File: WooCommerce/Classes/ViewModels/Order Details/OrderDetailsDataSource.swift
Comment (jaclync) [reply WooCommerce/Classes/ViewModels/Order Details/OrderDetailsDataSource.swift]: Makes sense! Updated in https://github.com/woocommerce/woocommerce-ios/pull/3769/commits/0125fea0e2b728d7ba5b1fb19a7ed1192a689d01

**Comment by jaclync** [comment]
@pmusolino would you like to take another look at the updated PR? 😄

**Comment by pmusolino** [comment]
> @pmusolino I think this misalignment is from the dynamic type to support various font sizes. If you change the system font size, the icon and text might look aligned in some sizes? This is a tricky issue, since the text could be any number of lines and I haven't found a good way to vertically align the center of the icon and the first line of the text. We also have an existing issue for this #3574 - I'll see how we can align this better separately!

This is right, but probably the text is aligned on top of the icon, so with small text sizes, it seems misaligned. If you align it at the center of the icon it should fix all the issues with all the text sizes.

Also with large titles there is a small vertical misalignment

**Comment by jaclync** [comment]
> Also with large titles there is a small vertical misalignment

@pmusolino did you mean the alignment here (just wanted to confirm, since I wasn't sure if by "large titles" you meant the recent navigation bar large titles change):



> This is right, but probably the text is aligned on top of the icon, so with small text sizes, it seems misaligned. If you align it at the center of the icon it should fix all the issues with all the text sizes.

The tricky part is that the icon height and text line height aren't always the same, so we can't easily align them vertically. Vertically aligning the icon and text at the center would be a design change all over the app. @woocommerce/mobile-designers do you think we're okay with this change? Like in the screenshot above, the icon and text (single or multiple lines) are center aligned vertically.

**Comment by pmusolino** [comment]
> @pmusolino did you mean the alignment here (just wanted to confirm, since I wasn't sure if by "large titles" you meant the recent navigation bar large titles change)

Yes exactly

> The tricky part is that the icon height and text line height aren't always the same, so we can't easily align them vertically. Vertically aligning the icon and text at the center would be a design change all over the app.

Right, but probably moving the label 4 pixel below, will improve the appearance with every text size. cc @woocommerce/mobile-designers

**Comment by Garance91540** [comment]
> @woocommerce/mobile-designers do you think we're okay with this change? Like in the screenshot above, the icon and text (single or multiple lines) are center aligned vertically.

Sounds good!

**Comment by jaclync** [comment]
> Right, but probably moving the label 4 pixel below, will improve the appearance with every text size.

@pmusolino Let's go with the vertical center alignment as G suggested! since the design change affects other parts of the app, I plan to work on it separately https://github.com/woocommerce/woocommerce-ios/issues/3574 for more thorough testing. Also, this PR would unblock other work like https://github.com/woocommerce/woocommerce-ios/issues/3279 and https://github.com/woocommerce/woocommerce-ios/pull/3785. Lemme know if you think there are other issues to address for this PR!

**Comment by jaclync** [comment]
@pmusolino do you think this PR is good to merge? (it hasn't got an approval from code review 😄 )

**Review by pmusolino — APPROVED** [review]
LGTM :shipit:

**Pull request merged**
